lib/apiservers/engine/backends/container.go

```diff
@@ -988,9 +990,25 @@ func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Con
 			ports = portInformation(t, ips)
 		}
 
+		// verify that the repo:tag exists for the container -- if it doesn't then we should present the
+		// truncated imageID -- if we have a failure determining then we'll show the data we have
+		repo := *t.ContainerConfig.RepoName
+		ref, _ := reference.ParseNamed(*t.ContainerConfig.RepoName)
+		if ref != nil {
+			imageID, err := cache.RepositoryCache().Get(ref)
+			if err != nil && err == cache.ErrDoesNotExist {
+				// the tag has been removed, so we need to show the truncated imageID
+				imageID = cache.RepositoryCache().GetImageID(*t.ContainerConfig.LayerID)
+				if imageID != "" {
+					id := uid.Parse(imageID)
+					repo = id.Truncate().String()
+				}
```

Comment:
No - docker will display <none> in the images output when pulled by digest.  In the ps output (which is this code) it will display the repo:tag, digest (repo:@sha256:blah) or trunc'd image id.  

Since we don't support pulling by digest yet, we cover the repo:tag & trunc'd id.